Improve nn.modules.activation and batchnorm docs

Jarlaze opened the pull request:
Fixes #112602

For some reason, I could not get the same output when running pycodestyle command as indicated in the issue. I manually ran ruff checks fixing the following issues  `D202`, `D204`,  `D205`, `D207`, `D400` and `D401`.

### Requested output

nn.modules.activation:
before: 135
after: 79

nn.modules.batchnorm 
before: 21
after: 3

Jarlaze committed:
Fix pydcostyle linting issues in activation.py

Fixed pydocstyle issues with running ruff check --select X, where X is error code from
pydocstyle actiovation.py --count.

Jarlaze committed:
Fix linting issues in batchnorm.py

Fix issues pointed by running command pydocstyle batchnorm.py --count

Jarlaze commented:
@pytorchbot label "docathon-h2-2023"

Jarlaze commented:
@pytorchbot label "topic: not user facing"

Jarlaze commented:
@pytorchbot label "topic: docs"

svekars commented:
You could still fix these in the actiovation.py: 
```
D205: 1 blank line required between summary line and description (found 0)
torch/nn/modules/activation.py:1077 in public method `forward`:
 D207: Docstring is under-indented
torch/nn/modules/activation.py:1077 in public method `forward`:
 D400: First line should end with a period (not ':')
```

svekars commented:
To run locally:
```
pip3 install pydocstyle
pydocstyle torch/nn/modules/activation.py --count
pydocstyle torch/nn/modules/batchnorm.py --count
```

Jarlaze commented:
@svekars Do you have a proposal for what the Multihead attention forward method docstring summary line should be? The D207 issue is easy to solve but the D400 requires generating a summary line. One way to fix the issue would be to change `Args:` -> `Args.` but that would not make too much sense from a good quality documentation perspective. 

I propose the summary line to be "Apply attention mechanism." Do you think this fix is reasonable?

Jarlaze commented:
@svekars Fixed issues. I gave a second thought to the forward method's summary line and made it more descriptive.

Jarlaze commented:
Hi @svekars, what is the status of this PR? Is there anything you would require from me or should we soon merge this?

mikaylagawarecki commented on torch/nn/modules/activation.py:
```diff
@@ -106,10 +108,9 @@ def extra_repr(self) -> str:
+    r"""Applies the randomized leaky rectified linear unit function, element-wise.
 
-    `Empirical Evaluation of Rectified Activations in Convolutional Network`_.
+    Method described in the paper:`Empirical Evaluation of Rectified Activations in Convolutional Network`_.
```
nit: not your fault but looks like this is missing the url of the paper https://arxiv.org/abs/1505.00853, could you add it please

mikaylagawarecki replied:
This is my bad, I just looked at the rendered public documentation on main and it seems that the paper is linked properly, so there's actually no need to change anything here

Jarlaze replied:
Deleted second mention of research paper.

mikaylagawarecki commented on torch/nn/modules/batchnorm.py:
```diff
@@ -421,8 +426,9 @@ def _check_input_dim(self, input):
-    the ``num_features`` argument of the :class:`BatchNorm2d` that is inferred
+    r"""A :class:`torch.nn.BatchNorm2d` module with lazy initialization.
+
+    Lazy initialization based on the the ``num_features`` argument of the :class:`BatchNorm2d` that is inferred
```
nit:
```suggestion
    Lazy initialization is done for the ``num_features`` argument of the :class:`BatchNorm2d` that is inferred
```

mikaylagawarecki commented on torch/nn/modules/batchnorm.py:
```diff
@@ -530,8 +537,9 @@ def _check_input_dim(self, input):
-    the ``num_features`` argument of the :class:`BatchNorm3d` that is inferred
+    r"""A :class:`torch.nn.BatchNorm3d` module with lazy initialization.
+
+    Lazy initialization based on the the ``num_features`` argument of the :class:`BatchNorm3d` that is inferred
```
nit:
```suggestion
    Lazy initialization is done for the ``num_features`` argument of the :class:`BatchNorm3d` that is inferred
```

mikaylagawarecki commented on torch/nn/modules/batchnorm.py:
```diff
@@ -563,8 +571,9 @@ def _check_input_dim(self, input):
-    with additional channel dimension) as described in the paper
+    r"""Applies Batch Normalization over a N-Dimensional input.
+
+    N-D a mini-batch of [N-2]D inputs with additional channel dimension) as described in the paper
```
nit:
```suggestion
    The N-D input is a mini-batch of [N-2]D inputs with additional channel dimension) as described in the paper
```

mikaylagawarecki commented on torch/nn/modules/batchnorm.py:
```diff
@@ -778,8 +787,7 @@ def forward(self, input: Tensor) -> Tensor:
     def convert_sync_batchnorm(cls, module, process_group=None):
-        r"""Helper function to convert all :attr:`BatchNorm*D` layers in the model to
-        :class:`torch.nn.SyncBatchNorm` layers.
+        r"""Convert all :attr:`BatchNorm*D` layers in the model to :class:`torch.nn.SyncBatchNorm` layers.
```
The initial phrasing looks ok to me, did pydocstyle complain here?

nit:
```suggestion
        r"""Converts all :attr:`BatchNorm*D` layers in the model to :class:`torch.nn.SyncBatchNorm` layers.
```

Jarlaze replied:
Yes, the issue is pydocstyle D401 error code "First line should be in imperative mood". `Helper` does not conform with this.

mikaylagawarecki commented on torch/nn/modules/activation.py:
```diff
@@ -1297,7 +1322,7 @@ def merge_masks(self, attn_mask: Optional[Tensor], key_padding_mask: Optional[Te
 
 class PReLU(Module):
-    r"""Applies the element-wise function:
+    r"""Applies the element-wise function.
```
```suggestion
    r"""Applies the element-wise PReLU function.
```

mikaylagawarecki commented on torch/nn/modules/activation.py:
```diff
@@ -1368,7 +1394,7 @@ def extra_repr(self) -> str:
 
 class Softsign(Module):
-    r"""Applies the element-wise function:
+    r"""Applies the element-wise function.
```
```suggestion
    r"""Applies the element-wise Softsign function.
```

mikaylagawarecki commented on torch/nn/modules/activation.py:
```diff
@@ -1391,7 +1417,7 @@ def forward(self, input: Tensor) -> Tensor:
 
 class Tanhshrink(Module):
-    r"""Applies the element-wise function:
+    r"""Applies the element-wise function.
```
```suggestion
    r"""Applies the element-wise Tanhshrink function.
```

mikaylagawarecki commented on torch/nn/modules/activation.py:
```diff
@@ -779,7 +798,7 @@ def extra_repr(self) -> str:
 
 class LogSigmoid(Module):
-    r"""Applies the element-wise function:
+    r"""Applies the element-wise function.
```
```suggestion
    r"""Applies the Logsigmoid function elementwise.
```

mikaylagawarecki replied:
There's one missing change here!

mikaylagawarecki commented on torch/nn/modules/activation.py:
```diff
@@ -728,7 +746,7 @@ def extra_repr(self) -> str:
 
 class LeakyReLU(Module):
-    r"""Applies the element-wise function:
+    r"""Applies the element-wise function.
```
```suggestion
    r"""Applies the LeakyReLU function elementwise.
```

mikaylagawarecki commented on torch/nn/modules/activation.py:
```diff
@@ -563,7 +576,7 @@ def extra_repr(self) -> str:
 
 class SELU(Module):
-    r"""Applied element-wise, as:
+    r"""Applied element-wise, as.
```
```suggestion
    r"""Applies the SELU function elementwise.
```

mikaylagawarecki commented on torch/nn/modules/activation.py:
```diff
@@ -519,7 +531,7 @@ def extra_repr(self) -> str:
 
 class CELU(Module):
-    r"""Applies the element-wise function:
+    r"""Applies the element-wise function.
```
```suggestion
    r"""Applies the CELU function elementwise.
```

mikaylagawarecki commented on torch/nn/modules/activation.py:
```diff
@@ -269,7 +272,7 @@ def extra_repr(self) -> str:
 
 class Sigmoid(Module):
-    r"""Applies the element-wise function:
+    r"""Applies the element-wise function.
```
```suggestion
    r"""Applies the Sigmoid function elementwise.
```

mikaylagawarecki commented on torch/nn/modules/activation.py:
```diff
@@ -239,7 +242,7 @@ def extra_repr(self) -> str:
 
 class ReLU6(Hardtanh):
-    r"""Applies the element-wise function:
+    r"""Applies the element-wise function.
```
```suggestion
    r"""Applies the ReLU6 function elementwise.
```

mikaylagawarecki reviewed (commented):
Thanks for the PR and many apologies for the delay in review!

Small ask: could you make sure that the spelling of elementwise is consistent please (whether you choose element-wise or elementwise). Thanks a lot!!

Jarlaze committed:
Update activation.py docstring

Fix function summary lines, improve wording, remove unnecessary research paper link.

Jarlaze committed:
Fix docstring wordings in batchnorm.py

Jarlaze commented:
No worries @mikaylagawarecki. I am just happy to help this awesome open-source community. I made the suggested edits. I left the discussion about missing URL from paper Empirical Evaluation of Rectified Activations in Convolutional Networ open, because there are now two URLs in the documentation pointing to the paper.

mikaylagawarecki reviewed (approved):
Thanks so much!

mikaylagawarecki commented:
@pytorchbot merge
